compiler/ballerina-parser/src/main/java/io/ballerina/compiler/internal/parser/BallerinaParserErrorHandler.java

```diff
@@ -3040,6 +3040,8 @@ private Result seekMatchInExpressionRhs(int lookahead, int currentDepth, int cur
             nextContext = ParserRuleContext.EQUALS_KEYWORD;
         } else if (parentCtx == ParserRuleContext.CLIENT_RESOURCE_ACCESS_ACTION) {
             nextContext = ParserRuleContext.CLOSE_BRACKET;
+        } else if (parentCtx == ParserRuleContext.CHECKING_EXPRESSION) {
+            nextContext = ParserRuleContext.ON_KEYWORD;
```

Comment:
@DamithaSenevirathne Seems this comment is not addressed. Now that we look at two tokens at the ballerina parser, we need to do the same in the error handler for the consistency. 
```java
            STToken nextToken = this.tokenReader.peek(lookahead);
            STToken nextNextToken = this.tokenReader.peek(lookahead + 1);
            if (nextToken.kind == SyntaxKind.ON_KEYWORD && nextNextToken.kind == SyntaxKind.FAIL_KEYWORD) {
                nextContext = ParserRuleContext.ON_FAIL_CHECK;
            } else {
                nextContext = ParserRuleContext.CHECK_EXPR_END;
            }
```
Having a separate variable is fine here for the readability. Otherwise, `this.tokenReader.peek(lookahead + 1)` could be confusing.